src/scene/layers/RenderLayer.ts

```diff
@@ -0,0 +1,267 @@
+ * ```js
+ * container.addChild(rect);    // Logical parent
+ * layer.attach(rect);      // Explicitly reassign to the layer
+ * ```
+ *
+ * ### 5. Layer Position in Scene Graph
+ * A layer's position in the scene graph determines its render priority relative to other layers and objects.
+ * Layers can be inserted anywhere in the scene graph.
+ * ```js
+ * const backgroundLayer = new RenderLayer();
+ * const uiLayer = new RenderLayer();
+ *
+ * app.stage.addChild(backgroundLayer);
+ * app.stage.addChild(world);
+ * app.stage.addChild(uiLayer);
+ * ```
+ * This is a new API and therefore considered experimental at this stage.
+ * While the core is pretty robust, there are still a few tricky issues we need to tackle.
+ * However, even with the known issues below, we believe this API is incredibly useful!
+ *
+ * Known issues:
+ *  - Interaction may not work as expected since hit testing does not account for the visual render order created by layers.
+ *    For example, if an object is visually moved to the front via a layer, hit testing will still use its original position.
+ *  - RenderLayers and their children must all belong to the same renderGroup to work correctly
+ */
+export class RenderLayerClass extends Container
+{
+    /**
+     * Default options for RenderLayer instances
+     * @property {boolean} sortableChildren - If true, layer children will be automatically sorted each render.
+     * Default false.
+     * @property {Function} sortFunction - Function used to sort layer children. Default sorts by zIndex.
+     */
+    public static defaultOptions: RenderLayerOptions = {
+        sortableChildren: false,
+        sortFunction: (a, b) => a.zIndex - b.zIndex,
+    };
+
+    /** Function used to sort layer children if sortableChildren is true */
+    public sortFunction: (a: Container, b: Container) => number;
+
+    /** List of objects to be rendered by this layer */
+    public renderLayerChildren: Container[] = [];
+    /**
+     * Creates a new RenderLayer instance
+     * @param options - Configuration options for the RenderLayer
+     * @param {boolean} [options.sortableChildren=false] - If true, layer children will be automatically sorted each render
+     * @param {Function} [options.sortFunction] - Custom function to sort layer children. Default sorts by zIndex
+     */
+    constructor(options: RenderLayerOptions = {})
+    {
+        options = { ...RenderLayerClass.defaultOptions, ...options };
+
+        super();
+
+        this.sortableChildren = options.sortableChildren;
+        this.sortFunction = options.sortFunction;
+    }
+
+    /**
+     * Add an object to this render layer. The object will be rendered as part of this layer
+     * while maintaining its original parent in the scene graph.
+     * If the object already belongs to a layer, it will be removed from the old layer before being added to this one.
+     * @param child - The object to add to this layer
+     */
+    public attach(child: Container)
```

Comment:
should this be the same API as `addChild` which accepts multiple children e.g `layer.attach(a, b, c)`